Mod2/Lessons/Week4/CRUDwithORM.md

```diff
@@ -0,0 +1,184 @@
+# CRUD with ORM
+
+## Intro
+
+Up to this point, you've created a database using PostGreSQL and pgAdmin. You've used CRUD operations to manipulate your database:
+
+-   (C)reate new data using the INSERT statement
+-   (R)ead existing data using the SELECT statement
+-   (U)pdate data using the UPDATE statement
+-   (D)elete data using the DELETE statement
+
+You've also used a `WHERE` clause to filter data, the `JOIN` keyword to combine data from multiple tables, and `ORDER BY` and `GROUP BY` to sort and arrange data for better analysis.
+
+In addition, you've used Entity Framework to create a database in PostGreSQL within a C# application. The next logical step is to CRUD-ify your database through your ORM. Is that a word? Do we have too many acronyms? Don't know yet, stay tuned.
+
+## Setup
+
+We will use the application and database from the Intro to ORM lesson and build from there.
+
+-   If you need a refresher, [revisit this lesson]() to build the application and database.    
+-   If you want a clean version of the application, [clone this repo]() and follow the instructions in the README.
+
+## Create
+
+In the previous lesson, we created the PlantTracker database with two tables: `plants` and `rooms`. There is a one-to-many relationship between rooms and plants: each plant is located in one room, and each room can have many plants. 
+
+Let's add a record to the room table. A room has an ID number (integer) as its primary key, along with a name field (text) and if the room has sunlight (boolean). Our first room is the kitchen that has ample sunlight.
+
+### Room
+
+Let's add a record to the room table. A room has an ID number as its primary key, along with a name field and if the room has sunlight. Our first room is the kitchen that has ample sunlight.
+
+> How would you add this room to your database using SQL statements?
+> `INSERT INTO rooms (id, name, has_sunlight)`
+> `VALUES (1, 'Kitchen', 'true');`
+
+In this example, we have given the room id a value of `1`. Remember that the `id` field automatically increments. This is the first room, so the `id` field of this record will (most likely) be given the value of `1`.
+
+> What happens if there is already a room with an id of `1` in our database?
+
+Let's use Entity Framework to perform this task for us. In `Program.cs`, add the following code in the `Main` method:
+
+  
+
+```c#
+using (var context = new PlantTrackerContext())
```

Comment:
Megan put a good note in my migrations pr where she was thinking maybe we should provide more context on exactly where in the file this should go because it looks like the first time they will see `using`. Your lesson will come before mine, so want to share this idea :)

Reply:
Does this work?
> Let's use Entity Framework to perform this task for us. In `Program.cs`, add the following code in the `Main` method: